From 1f97ce206cc74b0e3459b8aa0c72c0133b624c0d Mon Sep 17 00:00:00 2001
From: Ezio Lacandia Bijelkic <brabus84@gmail.com>
Date: Thu, 5 Oct 2017 15:41:52 +0200
Subject: [PATCH 34/42] Fix SystemUI FC after disabling navbar and unlocking
 the phone

to replicate: boot with navbar enabled, then disable navbar, then switch
the screen off, then unlock with fingeprint or power button

java.lang.IllegalArgumentException: View=com.android.systemui.statusbar.phone.NavigationBarFrame{e0f4d97 G.E...... ......ID 0,0-1080,129 #7f0a01c4 app:id/navigation_bar_frame} not attached to window manager

Change-Id: I5ae631c129af28615d26754848915e351970e422
---
 .../src/com/android/systemui/statusbar/phone/StatusBar.java    | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/phone/StatusBar.java b/packages/SystemUI/src/com/android/systemui/statusbar/phone/StatusBar.java
index 3bbcd4fc76d..8c2f9fde3d5 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/phone/StatusBar.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/phone/StatusBar.java
@@ -5893,8 +5893,11 @@ public class StatusBar extends SystemUI implements DemoMode, TunerService.Tunabl
             if (forcedVisibility && mNavigationBarView == null) {
                 createNavigationBar();
             } else if (mNavigationBarView != null) {
+                FragmentHostManager fm = FragmentHostManager.get(mNavigationBarView);
                 mWindowManager.removeViewImmediate(mNavigationBarView);
                 mNavigationBarView = null;
+                fm.getFragmentManager().beginTransaction().remove(mNavigationBar).commit();
+                mNavigationBar = null;
             }
         } else if (BERRY_GLOBAL_STYLE.equals(key)) {
             updateTheme();
-- 
2.17.1

